Fix error output if no callback is provided

bambamboole opened the pull request:
This PR fixes the error output if no callback is provided.

Instead of nesting if conditions I went for a early return approach. I also provided a unit test for the cases.

If you want me to change anything, just drop a line :)

maschmann commented:
Hi, thanks for contributing ;-) 
Normally I go for not having multiple returns within a method, but a single point of state return. There are some rare exceptions for this rule: e.g. if it has serious performance impacts due to heavy computational tasks. But if you have to consider such things, maybe the application design is flawed anyways :-D
In this particular case, it's OK to early return, though. Could you please fix the conflicts first?

bambamboole commented:
Fixed the conflicts and removed the just merged not used use statement.

I do not want to sell you on early returns :) 
As you said, in this case, it really makes sense, because we just have different cases and having them out of mind via an early return reduces cognitive overhead.

RoussKS commented:
Major fan of early returns here 😅😎

Doesn't matter, but helps for cleaner code imho

RoussKS commented on Tests/Asm/Ansible/Command/AnsiblePlaybookTest.php:
```diff
@@ -824,12 +818,6 @@ public function testExtraVars(): void
             ],
-            // Test valid JSON.
-            [
-                'input' => '{ "key1": "value1", "key2": "value2" }',
```
@maschmann @bambamboole I think that the json or empty string vars tests have been removed?
